src/sys/fuchsia/handles.rs

```diff
@@ -0,0 +1,66 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use magenta::HandleBase;
+use std::sync::Mutex;
+
+/// Wrapper for registering a `HandleBase` type with mio.
+#[derive(Debug)]
+pub struct EventedHandle<T> where T: HandleBase {
```

Comment:
Since this is a publicly exported structure under `mio::fuchsia`, could we trim this back a bit to make it more conservative? Perhaps make the fields private and maybe remove the generic for now?

(or is the generic necessary?)